rx_signal_processing/dsp.cu

```diff
@@ -883,39 +819,25 @@ void DSPCore::send_timing()
  */
 void DSPCore::cuda_postprocessing_callback(std::vector<double> freqs, uint32_t total_antennas,
                                             uint32_t num_samples_rf,
-                                            uint32_t num_output_samples_per_antenna_1,
-                                            uint32_t num_output_samples_per_antenna_2,
-                                            uint32_t num_output_samples_per_antenna_3,
-                                            uint32_t num_output_samples_per_antenna_4)
+                                            std::vector<uint32_t> samples_per_antenna,
+                                            std::vector<uint32_t> total_output_samples)
 {
-    #ifdef ENGINEERING_DEBUG
-      auto total_output_samples_1 = num_output_samples_per_antenna_1 * rx_freqs.size() *
-                                      total_antennas;
-      auto total_output_samples_2 = num_output_samples_per_antenna_2 * rx_freqs.size() *
-                                      total_antennas;
-      auto total_output_samples_3 = num_output_samples_per_antenna_3 * rx_freqs.size() *
-                                      total_antennas;
-      auto total_output_samples_4 = num_output_samples_per_antenna_4 * rx_freqs.size() *
-                                      total_antennas;
-
-      allocate_and_copy_first_stage_host(total_output_samples_1);
-      allocate_and_copy_second_stage_host(total_output_samples_2);
-      allocate_and_copy_third_stage_host(total_output_samples_3);
-      allocate_and_copy_fourth_stage_host(total_output_samples_4);
+  #ifdef ENGINEERING_DEBUG
+    for (uint32_t i=0; i<filter_outputs_d.size()-1; i++) {
+      allocate_and_copy_host(total_output_samples[i], filter_outputs_d[i]);
+    }
+  #endif
 
-    #endif
+  allocate_and_copy_host(total_output_samples.back(), filter_outputs_d.back());
 
-    rx_freqs = freqs;
-    num_rf_samples = num_samples_rf;
-    num_antennas = total_antennas;
-    num_first_stage_samples_per_antenna = num_output_samples_per_antenna_1;
-    num_second_stage_samples_per_antenna = num_output_samples_per_antenna_2;
-    num_third_stage_samples_per_antenna = num_output_samples_per_antenna_3;
-    num_fourth_stage_samples_per_antenna = num_output_samples_per_antenna_4;
+  rx_freqs = freqs;
```

Comment:
You don't need this or self in c++ to access class members. it does have a this identifier that I use in here since I couldn't think of a different name to use.